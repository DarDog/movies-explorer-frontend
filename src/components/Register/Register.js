import React from 'react';
import './Register.css'

const Register = () => {
  return (
    <form className='register' name='register'>
      <div className="register__inputs">
        <label className='register__label'>Имя</label>
        <input className='register__input' type="text"/>
        <span className='register__error'>{}</span>
        <label className='register__label'>E-mail</label>
        <input className='register__input register__input_status_correct' type="email"/>
        <span className='register__error'>{}</span>
        <label className='register__label'>Пароль</label>
        <input className='register__input register__input_status_error' type="password"/>
        <span className='register__error'>Не пароль</span>
      </div>
      <button className="register__submit">Зарегистрироваться</button>
    </form>
  );
}

export default Register;
