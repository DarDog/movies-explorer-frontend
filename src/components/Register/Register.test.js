import React from 'react';
import Register from "./Register";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import '@testing-library/jest-dom';

describe('Компонент Register',() => {
  it('Компонент рендерится', () => {
    render(
      <BrowserRouter>
        <Register/>
      </BrowserRouter>
    );

    expect(screen.getByRole('form')).toBeInTheDocument();
    expect(screen.getAllByRole('textbox')).toHaveLength(2);
    expect(screen.getByRole('button')).toBeInTheDocument();
  })
  it('Соответствует своему snapshot', () => {
    const component =  render(
      <BrowserRouter>
        <Register/>
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  })
})
