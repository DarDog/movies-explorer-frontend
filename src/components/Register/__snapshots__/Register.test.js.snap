// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Register Соответствует своему snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="register"
        name="register"
        novalidate=""
      >
        <div
          class="register__inputs"
        >
          <label
            class="register__label"
          >
            Имя
          </label>
          <input
            class="register__input register__input_status_error"
            maxlength="30"
            minlength="2"
            required=""
            type="text"
            value=""
          />
          <span
            class="register__error"
          />
          <label
            class="register__label"
          >
            E-mail
          </label>
          <input
            class="register__input register__input_status_error"
            required=""
            type="email"
            value=""
          />
          <span
            class="register__error"
          />
          <label
            class="register__label"
          >
            Пароль
          </label>
          <input
            class="register__input register__input_status_error"
            maxlength="30"
            minlength="8"
            required=""
            type="password"
            value=""
          />
          <span
            class="register__error"
          />
        </div>
        <button
          class="register__submit"
        >
          Зарегистрироваться
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="register"
      name="register"
      novalidate=""
    >
      <div
        class="register__inputs"
      >
        <label
          class="register__label"
        >
          Имя
        </label>
        <input
          class="register__input register__input_status_error"
          maxlength="30"
          minlength="2"
          required=""
          type="text"
          value=""
        />
        <span
          class="register__error"
        />
        <label
          class="register__label"
        >
          E-mail
        </label>
        <input
          class="register__input register__input_status_error"
          required=""
          type="email"
          value=""
        />
        <span
          class="register__error"
        />
        <label
          class="register__label"
        >
          Пароль
        </label>
        <input
          class="register__input register__input_status_error"
          maxlength="30"
          minlength="8"
          required=""
          type="password"
          value=""
        />
        <span
          class="register__error"
        />
      </div>
      <button
        class="register__submit"
      >
        Зарегистрироваться
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
